### ISSUE TITLE
simplified occlusion implementation 

### ISSUE THREAD

**Issue by fawazsammani** [pytorch/captum#1022]
Hello. I would like to know how exactly you are implementing the Occlusion algorithm. It's quite hard to follow what's happening from the library source code, so i was following the reference you pointed out [here](https://github.com/marcoancona/DeepExplain/blob/master/deepexplain/tensorflow/methods.py). I am getting very different results between the library and my own implementation following the reference you posted. 

My input is a text that is mapped to ids, of shape `(1,7)` where 7 are the number of subtokens. 

Using the library, the code is as follows:

```
def forward_func(input_ids):
    
    predictions = model(input_ids)

    return predictions.logits

_, label = torch.max(model(input_ids).logits, dim=-1)
occlusion_ablator = Occlusion(forward_func)
attributions = occlusion_ablator.attribute(inputs=input_ids,
                                           sliding_window_shapes=(1,),
                                           target = label)
# value to replace is 0 by default
```
Using my own implementation, the code is as follows:

```
_, label = torch.max(model(input_ids).logits, dim=-1)
def get_score(input_ids, label):
    
    predictions = model(input_ids)

    score = predictions.logits[:, label.item()]
    return score

initial_eval_score = get_score(input_ids).detach()

attributions = torch.zeros_like(input_ids)
for i in range(input_ids[0].shape[0]):
    t_input = input_ids.detach().clone()
    t_input[0][i] = 0
    score = get_score(t_input)
    attributions[0][i] = score.detach()
    
attributions = initial_eval_score - attributions
# Weight matrix to divide the attributions by is not needed since there are no overlaps (stride = 1 and sliding_window=1)
```

May I please know what is wrong?

**Comment by fawazsammani** [comment]
Solved. 
Stupid mistake was `torch.zeros_like(input_ids)` where the tensor type was Long, and was thus rounding up the logits.
